runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,585 @@
+		return provider.FatalError{Err: err}
+	}
+	err = os.Remove(path.Dir(scanConfigFileName))
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to remove scan config file. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	containerId, err := c.getContainerIdFromContainerName(ctx, scanName)
+	if err != nil {
+		return provider.FatalError{Err: err}
+	}
+	err = c.dockerClient.ContainerRemove(ctx, containerId, types.ContainerRemoveOptions{})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to remove scan container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	return nil
+}
+
+func (c Client) getContainerImageInfoFromImage(ctx context.Context, id string) (models.ContainerImageInfo, error) {
+	i, _, err := c.dockerClient.ImageInspectWithRaw(ctx, id)
+	if err != nil {
+		return models.ContainerImageInfo{}, fmt.Errorf("failed to get image. Provider=%s: %w", models.Docker, err)
+	}
+
+	return models.ContainerImageInfo{
+		Architecture: utils.PointerTo(i.Architecture),
+		Id:           utils.PointerTo(i.ID),
+		Labels:       convertTags(i.Config.Labels),
+		Name:         utils.PointerTo(i.Config.Image),
+		ObjectType:   "ContainerImageInfo",
+		Os:           utils.PointerTo(i.Os),
+		Size:         utils.PointerTo(int(i.Size)),
+	}, err
+}
+
+func (c Client) getAssetFromImage(ctx context.Context, id string) (models.AssetType, error) {
+	asset := models.AssetType{}
+
+	info, err := c.getContainerImageInfoFromImage(ctx, id)
+	if err != nil {
+		return asset, err
+	}
+
+	err = asset.FromContainerImageInfo(info)
+	if err != nil {
+		return asset, fmt.Errorf("failed to create AssetType from ContainerImageInfo. Provider=%s: %w", models.Docker, err)
+	}
+
+	return asset, err
+}
+
+func convertTags(tags map[string]string) *[]models.Tag {
+	ret := make([]models.Tag, 0, len(tags))
+	for key, val := range tags {
+		ret = append(ret, models.Tag{
+			Key:   key,
+			Value: val,
+		})
+	}
+	return &ret
+}
+
+func getScanName(config *provider.ScanJobConfig) (string, error) {
+	id, err := getAssetId(config)
+	if err != nil {
+		return "", provider.FatalError{
+			fmt.Errorf("failed to get scan name. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	return "scan_" + config.ScanID + "_asset_" + strings.Replace(id, ":", "_", -1), nil
+}
+
+func getScanConfigFileName(config *provider.ScanJobConfig) (string, error) {
+	scanName, err := getScanName(config)
+	if err != nil {
+		return "", provider.FatalError{Err: err}
+	}
+
+	// TODO (adamtagscherer): check if os.TempDir() doesn't create a new unique directory every time it's being called
+	return os.TempDir() + scanName + "_scanconfig.yaml", nil
+}
+
+func getAssetId(config *provider.ScanJobConfig) (string, error) {
+	objectType, err := config.AssetInfo.Discriminator()
+	if err != nil {
+		return "", provider.FatalError{
+			Err: fmt.Errorf("failed to get asset object type. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	switch objectType {
+	case "ContainerInfo":
```

Comment:
I would suggest creating an internal type e.g. `type dockerAssetType string` and defining ContainerInfor and ContainerImageInfo as internal constants.

Reply:
solved with https://github.com/openclarity/vmclarity/pull/462#discussion_r1265406115